### PR TITLE
JS Hook

### PR DESCRIPTION
**Pull request by joeybaker**
`data-js="login"` vs. the old `class="js-login"`. See commit for reasons.

### CODE REVIEW

**Comment by tedbreen** [comment]
I have come to appreciate the `data-js` attribute. I see it as serving two purposes:
* it indicates that something will be appended here by means of JavaScript/Backbone.
* it strongly indicates that this attribute will NOT be used for CSS styling.

Even if the `js-` prefix indicates "DON'T USE FOR STYLING," I think it is messy and possibly confusing if an HTML element has additional classes that will be used for CSS styling. (for example, `<div class="btn orange fullWidth js-btn">`)

File: README.md [hunk 1]
```diff
@@ -22,7 +22,7 @@ This style guide is based on the structure provided by @fat in [Medium's Style G
   - [ComponentName](#componentname)
   - [componentName--modifierName](#componentname--modifiername)
   - [componentName-descendantName](#componentname-descendantname)
-  - [componentName.is-stateOfComponent](#componentname.is-stateofcomponent)
+  - [componentName.is-stateOfComponent](#componentnameis-stateofcomponent)
```
Comment (CristiScheye) [inline README.md]: What's the use case for this (or the deleted version)? How does this differ from the `--modifierName`?

Comment (joeybaker) [inline README.md]: Agreed. The `is-*` felt redundant to me too.

**Comment by joeybaker** [comment]
Closing in favor of #3

**Pull request closed without merging**